sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/MetricStorageUtils.java

```diff
@@ -24,6 +24,9 @@ private MetricStorageUtils() {}
    */
   static <T> void mergeInPlace(
       Map<Attributes, T> result, Map<Attributes, T> toMerge, Aggregator<T> aggregator) {
+    if (result == null || toMerge == null) {
```

Comment:
I dont think these checks are required anymore after the updates to remove `@Nullable` from `TemporalMetricStorage#LastReportedAccumulation`.